crates/shared/src/lib.rs

```diff
@@ -135,12 +135,20 @@ macro_rules! shared_api {
         }
 
         struct Function<'a> {
-            arg_names: Vec<String>,
+            args: Vec<FunctionArgumentData>,
             asyncness: bool,
             name: &'a str,
             generate_typescript: bool,
             generate_jsdoc: bool,
             variadic: bool,
+            ret_ty_override: Option<String>,
+            ret_desc: Option<String>,
+        }
+
+        struct FunctionArgumentData {
+            name: String,
+            ty_override: Option<String>,
+            desc: Option<String>,
```

Comment:
```suggestion
        struct FunctionArgumentData<'a> {
            name: String,
            ty_override: Option<&'a str>,
            desc: Option<&'a str>,
```
This will not make it work in `AuxExport` anymore, so we need a new type there. E.g. `AuxFunctionArgumentData`.

Reply:
is this necessary/needed? we can just use shared `FunctionArgumentData` here, instead of just replicating it as Aux again

Reply:
Yes, we are reducing allocations and keeping consistent with the existing code. Ideally, we avoid allocations even in `AuxExport`, but I assume that would require a much bigger refactor.

FWIW: In big projects `wasm-bindgen` can get quite slow, I've seen up to 12s to post-process locally (on an Intel 13900K). The biggest bottleneck we can do anything about are allocations. While these won't save us any significant time, we shouldn't start making changes that make the situations actively worse either.

Reply:
understood, I'll apply the chnages